nvme-print-stdout.c

```diff
@@ -1415,174 +1443,200 @@ static void stdout_registers_pmrswtp(__u32 pmrswtp)
+}
 
-			printf("cmbmsc	: %"PRIx64"\n", cmbmsc);
-			stdout_registers_cmbmsc(cmbmsc);
+void stdout_ctrl_registers(void *bar, bool fabrics)
+{
+	uint64_t cap = mmio_read64(bar + NVME_REG_CAP);
+	uint32_t vs = mmio_read32(bar + NVME_REG_VS);
+	uint32_t cc = mmio_read32(bar + NVME_REG_CC);
+	uint32_t csts = mmio_read32(bar + NVME_REG_CSTS);
+	uint32_t nssr = mmio_read32(bar + NVME_REG_NSSR);
+	uint32_t crto = mmio_read32(bar + NVME_REG_CRTO);
+	uint32_t cmbsz = mmio_read32(bar + NVME_REG_CMBSZ);
+	uint32_t pmrctl = mmio_read32(bar + NVME_REG_PMRCTL);
+	bool human = stdout_print_ops.flags & VERBOSE ? true : false;
 
-			printf("cmbsts	: %x\n", cmbsts);
-			stdout_registers_cmbsts(cmbsts);
+	if (cap != -1)
+		stdout_ctrl_register_support(bar, NVME_REG_CAP, human, true);
 
-			printf("pmrcap  : %x\n", pmrcap);
-			stdout_registers_pmrcap(pmrcap);
+	if (vs != -1)
+		stdout_ctrl_register_support(bar, NVME_REG_VS, human, true);
 
-			printf("pmrctl  : %x\n", pmrctl);
-			stdout_registers_pmrctl(pmrctl);
+	if (cc != -1)
+		stdout_ctrl_register_support(bar, NVME_REG_CC, human, true);
 
-			printf("pmrsts  : %x\n", pmrsts);
-			stdout_registers_pmrsts(pmrsts, pmrctl);
+	if (csts != -1)
+		stdout_ctrl_register_support(bar, NVME_REG_CSTS, human, true);
 
-			printf("pmrebs  : %x\n", pmrebs);
-			stdout_registers_pmrebs(pmrebs);
+	if (nssr != -1)
+		stdout_ctrl_register_support(bar, NVME_REG_NSSR, human, true);
 
-			printf("pmrswtp : %x\n", pmrswtp);
-			stdout_registers_pmrswtp(pmrswtp);
+	if (crto != -1)
+		stdout_ctrl_register_support(bar, NVME_REG_CRTO, human, true);
 
-			printf("pmrmscl	: %#x\n", pmrmscl);
-			stdout_registers_pmrmscl(pmrmscl);
+	if (fabrics)
+		return;
 
-			printf("pmrmscu	: %#x\n", pmrmscu);
-			stdout_registers_pmrmscu(pmrmscu);
-		}
-	} else {
-		if (cap != 0xffffffff)
-			printf("cap     : %"PRIx64"\n", cap);
-		if (vs != 0xffffffff)
-			printf("version : %x\n", vs);
-		if (cc != 0xffffffff)
-			printf("cc      : %x\n", cc);
-		if (csts != 0xffffffff)
-			printf("csts    : %x\n", csts);
-		if (nssr != 0xffffffff)
-			printf("nssr    : %x\n", nssr);
-		if (crto != 0xffffffff)
-			printf("crto    : %x\n", crto);
-		if (!fabrics) {
-			printf("intms   : %x\n", intms);
-			printf("intmc   : %x\n", intmc);
-			printf("aqa     : %x\n", aqa);
-			printf("asq     : %"PRIx64"\n", asq);
-			printf("acq     : %"PRIx64"\n", acq);
-			printf("cmbloc  : %x\n", cmbloc);
-			printf("cmbsz   : %x\n", cmbsz);
-			printf("bpinfo  : %x\n", bpinfo);
-			printf("bprsel  : %x\n", bprsel);
-			printf("bpmbl   : %"PRIx64"\n", bpmbl);
-			printf("cmbmsc	: %"PRIx64"\n", cmbmsc);
-			printf("cmbsts	: %x\n", cmbsts);
-			printf("pmrcap  : %x\n", pmrcap);
-			printf("pmrctl  : %x\n", pmrctl);
-			printf("pmrsts  : %x\n", pmrsts);
-			printf("pmrebs  : %x\n", pmrebs);
-			printf("pmrswtp : %x\n", pmrswtp);
-			printf("pmrmscl	: %#x\n", pmrmscl);
-			printf("pmrmscu	: %#x\n", pmrmscu);
-		}
-	}
+	stdout_ctrl_register_support(bar, NVME_REG_INTMS, human, true);
```

Comment:
Since only for the fabrics flag set case the values gotten by calling `nvme_get_properties()` and also the value is set -1 if failed `nvme_get_property()` in the function as below so just skip the output for the fabrics -1 value by the show-regs command.
```c
static int nvme_get_properties(int fd, void **pbar)
{
...
	memset(bar, 0xff, size);
	for (offset = NVME_REG_CAP; offset <= NVME_REG_CMBSZ;) {
...
		err = nvme_get_property(&args);
		if (nvme_status_equals(err, NVME_STATUS_TYPE_NVME, NVME_SC_INVALID_FIELD)) {
			err = 0;
			value = -1;
...
		if (nvme_is_64bit_reg(offset)) {
			*(uint64_t *)(bar + offset) = value;
...
		} else {
			*(uint32_t *)(bar + offset) = value;
...
		}
```